krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -106,60 +106,75 @@ func (r *itfce) GetGoStruct() (*nephioreqv1alpha1.Interface, error) {
 // GetAttachmentType returns the attachmentType from the spec
 // if an error occurs or the attribute is not present an empty string is returned
 func (r *itfce) GetAttachmentType() string {
+	if r.o == nil {
```

Comment:
I think this was using a common function. This block is very common for all getters
{
if r.o == nil {
		return ""
	}
	return r.getStringValue(foo...)
}

My suggestion is to handle the nil condition in common function itself.

Reply:
FYI There was code being processed earlier without checking for the nil condition and had mentioned this to Wim. I made the point that if check for nil fails, no need to do further processing. @gvbalaji Maybe, it would be a good idea for you, Wim and I to discuss this quick to avoid further confusion? What do you say?

Reply:
With the current flow we valid input first and do nil check after if we apply to the object. At this stage the code is most optimal.